Test/gui

FarisElghlan opened the pull request:
Close #59 
Tested most of the GUI classes.

Please only merge this branch after #57 has been reviewed and merged, since this branch depends on the branch of PR #57. Only the classes in the test package have to be reviewed, since the other changed will be reviewed at PR #57 already.

wouterjsmit committed:
Add tests

The gui is now 27.12% test covered.
Tested the following classes:
- ViewNode (97.39%)
- Area (100%)
- AnimationEvent (100%)
- GraphicsChangedEvent (100%)

FarisElghlan committed:
Added node density heatmap tests

87% coverage (everything except for the animation)

FarisElghlan committed:
Tested TextDescription and HeatmapManager

FarisElghlan committed:
Fix failing build on headless environment

The last travis build failed because travis can't open a display (so it can't launch JavaFX).
I have tried to turn off the JavaFX tests just for travis and can only test if it succeeded by pushing.

FarisElghlan committed:
Tested the other GUI classes

Tested:
- DrawGraph (a little bit, because it will change a lot in the future).
- DescriptionPane
- SelectionManager
- TreeManager (missing a lot of coverage because of animations).

FarisElghlan commented:
## Test coverage (line coverage) of the GUI module:

![coverage](https://cloud.githubusercontent.com/assets/9753699/15126447/613b5dd0-1632-11e6-8a95-f802f1c188b6.png)
See #59 for more details

wouterjsmit commented:
Fixed the conflicts that were caused by a rebase in #57. The remaining commits are correctly only the ones that are relevant to this PR

wouterjsmit commented on PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionManagerTest.java:
```diff
@@ -0,0 +1,98 @@
+   */
+  @Test
+  public void testSelect() {
+    Pane pane = new Pane();
```
You could move the variable initialization/declaration to a `@Before` method

wouterjsmit commented on PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/view/tree/TreeManagerTest.java:
```diff
@@ -0,0 +1,165 @@
+   */
+  @Test
+  public void testSetOnLeavesChanged() {
+    SelectionManager selectionManager = new SelectionManager(new Pane(), new Pane());
```
You could also move this variable declaration to the `@Before` method.

wouterjsmit commented on PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/view/tree/ViewNodeTest.java:
```diff
@@ -0,0 +1,266 @@
+   */
+  @Test
+  public void testSelect() {
+    // NOT IMPLEMENTED YET
```
Not implemented?

FarisElghlan replied:
In the code this method is not implemented yet. The method does however exist, as it is inherited from an interface. This method will make some visual changes to the UI when a node is selected in the future (ex. make the selected node bigger or a different color). I kept the test method because I was too lazy too delete it and later add it when the method is implemented.

wouterjsmit commented on PL2/PL2-gui/src/test/java/nl/tudelft/pl2016gr2/gui/view/tree/heatmap/NodeDensityHeatmapTest.java:
```diff
@@ -0,0 +1,81 @@
+import java.util.ArrayList;
+
+/**
+ * This class tests the {@link ViewNode} class.
```
Typo `ViewNode -> NodeDensityHeatMap`

FarisElghlan replied:
Nice catch. I should probably add this line to my test template so it gets automatically generated.

wouterjsmit commented on PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/test/utility/AccessPrivate.java:
```diff
@@ -21,12 +21,6 @@
 public final class AccessPrivate {
 
   /**
-   * Don't let anyone instantiate this class.
```
We should just exclude these annotation files from the coverage report (and all analysis tools), because they aren't part of our project.

Keep the constructor, it provides safety.

FarisElghlan replied:
Ok I did not know that was possible, that would indeed be a good idea!
All methods of the class are however static, so being able to instantiate the class isn't really a problem.

And to review your review: could you add  you comments to the end of the section which they apply to? So in this case to the line containing the `}` (line 27) below. Then I can see the full code section which you comment applies to (if it isn't too big) in the pull request messages, instead of having to check the full code changes.

wouterjsmit replied:
Thanks for the tip, didn't think about that.

See #65 for the exclusion of your files. I suggest that you remove all commits that perform any (style) changes to these files. (please do not revert them, as then we not only have 1 unwanted commit, but 2)

`git rebase -i <some point in history>` and remove the commit with these changes.

If a commit accidentally also contained actual work (on top of the changes to annotation classes), then you can split it by changing `pick` to `edit`. You will stop at the state just before commiting that commit and will be able to reset your index with `git reset HEAD~1`. then you can re do that commit and modify any of your changes to either commit or leave them

wouterjsmit commented:
@FarisElghlan waiting for your feedback on the `@Before` declaration comments and on the style changes in the annotation classes which should/could be removed from the commits

FarisElghlan committed:
Fixed failing build

Build failed because a graph which was used for testing was deleted.
Now using a different test graph.

FarisElghlan committed:
Code cleanup

Adjusted the code as requested by @Pathemeous by moving variable
initialization to the `@ Before` method. Also fixed the incorrect class
reference.

FarisElghlan committed:
Merge branch 'dev' into test/gui

Conflicts:
	PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/tree/heatmap/NodeDensityHeatmap.java

FarisElghlan commented:
Resolved/answered all of your code review feedback. I think this branch is ready for merging.
